### DISCUSSION TITLE
Exclude personal settings from git code

### DISCUSSION THREAD

**Discussion by dfluegel** [Gifford47/HCPBridgeMqtt#20] [Ideas]
Hi,

meanwhile my first garage door is working fine (thanks @Gifford47 @14yannick for your help!), and I've ordered a few PCBs with the design for the first and the second garage, because my prototype is too big for the little box.

I've seen, that this project is a fork of a fork of a fork from the hkiam-project, there were some sensitive informations stored in an include file outside of the git path. Was it intended to move the SSID/PW config back into the git file? If not, I would like to change it (And also with a little enhancement, so the name of the ESP/MQTT device could be changed, so I would be able to control both of my garage doors ;-))

**Comment by Gifford47** [comment]
Hi!

Yes my intention was to make it more beginner friendly. So you only need to modify your credentials and start compiling. Currently I see no advantage to split up the configuration parameters into multiple files. The future plan is to use the WiFimanager again, so we don't need the hard coded wifi credentials anymore.
The best afford would be to build the hardware, flash the standard firmware and configure any private parameters at runtime (you would also be able to change the WiFi network without re-flashing esp).

**Comment by dfluegel** [comment]
This sounds like a cool idea.

Do you have also an idea or a plan, how to configure a second garage door without affecting the first one when both are online?

**Reply by Gifford47** [reply]
you only need to edit the base topic in [configuration.h ](https://github.com/Gifford47/HCPBridgeMqtt/blob/45d74e2cfd496078a9e5549246c36dab390a0cc1/HCPBridgeESP32/src/configuration.h#L54C5-L54C42) with your new topic for the second garage door.
F.e.:
```#define FTOPIC "hormann/garage_door2" ```

**Reply by 14yannick** [reply]
Needs the name in the [main.cpp](https://github.com/Gifford47/HCPBridgeMqtt/blob/master/HCPBridgeESP32/src/main.cpp) lin 440 and 441 also be changed.
```
  device["identifiers"] = "Garage Door 2";
  device["name"] = "Garage Door 2";
```
I just go trough the [mqtt documentation](https://www.home-assistant.io/integrations/mqtt/#discovery-messages) and by my understanding the mqtt device should be defined as following
```
   JsonArray identifiers= device.createNestedArray("identifiers");
   identifiers.add("Garage Door 2");
   device["name"] = "Garage Door 2";
```
as a list is expected for the attribute identifiers.
I will do a short test this evening and then create a variable in the config file for the name and id of the garage door.

**Reply by dfluegel** [reply]
I've tried editing (only) the lines in the main.pp, but with no good result. I'm going to try changing these lines in the main.pp and the FTOPIC.

**Reply by dfluegel** [reply]
Seems to work fine, the existing garage bridge is still online as the new one comes up. Thanks for your help again ;-)

**Reply by dfluegel** [reply]
Or maybe not... After the second ESP is online, my first ESP went offline / not available

![grafik](https://github.com/Gifford47/HCPBridgeMqtt/assets/64017393/acdcbdc7-269c-4baa-9fa7-0577d4ba7b3a)

**Reply by 14yannick** [reply]
I just add two parameters in the config file in my [dev Branche ](https://github.com/Gifford47/HCPBridgeMqtt/tree/14yannick_dev) 
See line 15 and 16 of the [configuration.h](https://github.com/Gifford47/HCPBridgeMqtt/blob/14yannick_dev/HCPBridgeESP32/src/configuration.h) file. 
```
    const char DEVICENAME[] = "Garage Door";
    #define DEVICE_ID  "garage_door"
```
I haven't tested it yet but you could have a look there.

**Reply by Gifford47** [reply]
@14yannick the [development ](https://github.com/Gifford47/HCPBridgeMqtt/tree/development) branch is a copy from your 14yannick_dev branch without the last three commits. Do you agree to only use the development branch in the future and put the 14yannick_dev into archive?

**Comment by 14yannick** [comment]
Hi @dfluegel have you been able to run our two doors with the last addition I have done?
I will start with the implementations to allow the change of this variables in the web interface. In a second step this will allow to change the MQTT setup information without modifying the configuration.h file

**Reply by dfluegel** [reply]
Hi, today i've recieved the PCBs - I wanted to wait, until i'm able to have both garage door controllers in a valid, state - currently the second controller is currently "just" the ESP32. Unfortunately i've ordered the wrong RJ11 connectors and i'm currently having the wrong esp32 board for the ordered PCB...

After making sure, that all these things are fixed, I'll get more reliable test results without stange behaviours.

**Reply by 14yannick** [reply]
What esp32 and connector you need. I'm starting to get a lot of spare parts maybe I can mail you the missing parts

**Reply by dfluegel** [reply]
Thanks a lot. Currently i need the matching ESP32 board for your PCB layout - and the RJ12 connector. Because I'm quite impatient, I've already ordered both, the correct ESP32 with the matching pin layout (same as here ESP32-DivKitC Pro PCB 2D.png) should arrive from China within the next 12 days.

To turn the tide ;-):
I've ordered quite a bunch of the (blue) PCBs, so if anyone else needs a PCB with vladimirzrnic design, please let me know. (It's just the blank PCB, but the other components would be required anyways).

Edit: I've corrected the name with the other PCB design.

**Reply by 14yannick** [reply]
You assembling this board? You are in Germany right? If you're interested send me a message in the home assistant community
![16904942070853165913999497452916](https://github.com/Gifford47/HCPBridgeMqtt/assets/6718914/709f7c24-2429-4993-9065-66317881ac1b)

**Reply by dfluegel** [reply]
Yes, I'm assembling it by myself (and just for me) and I'm in Germany. 

Correction of my post above: I've ordered the other, because it should fit into the little box once I've the correct components.

For now I'm not able to write to you in the HA community, because I've a new account. Almost all of my problems/topics I was able by just reading posts, so there wasn't a need to register until now.
